src/dsp_tools/commands/xmlupload/make_rdf_graph/make_values.py

```diff
@@ -51,157 +69,142 @@ def make_values(properties: list[XMLProperty], res_bnode: BNode, lookups: Lookup
-    permissions_lookup: dict[str, Permissions],
-) -> Graph:
+    res_bn: BNode,
+) -> tuple[BNode, Graph]:
+    val_bn = BNode()
+    g = _add_optional_triples(val_bn, val.permissions, val.comment)
+    g.add((res_bn, URIRef(val.prop_iri), val_bn))
+    g.add((val_bn, RDF.type, prop_type_info.knora_type))
+    return val_bn, g
+
+
+def _add_optional_triples(val_bn: BNode, permissions: Permissions | None, comment: str | None) -> Graph:
     g = Graph()
-    for val in prop.values:
-        transformed_val = transformer(val.value)
-        resolved_permission = resolve_permission(val.permissions, permissions_lookup)
-        transformed = TransformedValue(
-            value=transformed_val, prop_name=prop_name, permissions=resolved_permission, comment=val.comment
-        )
-        g += _make_simple_value_graph(transformed, res_bn, prop_type_info)
+    if permissions:
+        g.add((val_bn, KNORA_API.hasPermissions, Literal(str(permissions), datatype=XSD.string)))
+    if comment:
+        g.add((val_bn, KNORA_API.valueHasComment, Literal(comment, datatype=XSD.string)))
     return g
 
 
-def _make_simple_value_graph(
-    val: TransformedValue,
-    res_bn: BNode,
+def _make_value_graph_with_literal_object(
+    val: LiteralValueTypesAlias,
     prop_type_info: RDFPropTypeInfo,
+    res_bn: BNode,
 ) -> Graph:
-    val_bn = BNode()
-    g = _add_optional_triples(val_bn, val.permissions, val.comment)
-    g.add((res_bn, val.prop_name, val_bn))
-    g.add((val_bn, RDF.type, prop_type_info.knora_type))
-    g.add((val_bn, prop_type_info.knora_prop, val.value))
+    val_bn, g = _make_base_value_graph(val, prop_type_info, res_bn)
+    g.add((val_bn, prop_type_info.knora_prop, Literal(val.value, datatype=prop_type_info.xsd_type)))
     return g
 
 
-def _make_list_prop_graph(
-    prop: XMLProperty,
+def _make_list_value_graph(
+    val: IntermediaryList,
     res_bn: BNode,
-    prop_name: URIRef,
-    permissions_lookup: dict[str, Permissions],
-    listnode_lookup: dict[str, str],
+    prop_type_info: RDFPropTypeInfo,
 ) -> Graph:
-    g = Graph()
-    for val in prop.values:
-        s = assert_is_string(val.value)
-        if not (iri_str := listnode_lookup.get(s)):
-            msg = (
-                f"Could not resolve list node ID '{s}' to IRI. "
-                f"This is probably because the list node '{s}' does not exist on the server."
-            )
-            raise BaseError(msg)
-
-        resolved_permission = resolve_permission(val.permissions, permissions_lookup)
-        transformed = TransformedValue(
-            value=URIRef(iri_str),
-            prop_name=prop_name,
-            permissions=resolved_permission,
-            comment=val.comment,
-        )
-        g += _make_simple_value_graph(transformed, res_bn, RDF_PROP_TYPE_MAPPER["list"])
+    val_bn, g = _make_base_value_graph(val, prop_type_info, res_bn)
+    g.add((val_bn, prop_type_info.knora_prop, URIRef(val.value)))
```

Comment:
`URIRef` is very different from `xsd` therefore I prefer to do it separately from the values that are `Literals`. It would be possible in a generic way, but I think it would make more work in total.